src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.Node.cs

```diff
@@ -72,5 +78,87 @@ private string GetDebuggerDisplay()
                 return NameSpan + ", " + ParentIndex;
             }
         }
+
+        public readonly struct ParameterTypeInfo
+        {
+            public string Name { get; }
+            public bool IsComplexType { get; }
+
+            public ParameterTypeInfo(string name, bool isComplex)
+            {
+                Name = name;
+                IsComplexType = isComplex;
+            }
+
+            public override string ToString()
+                => $"{Name}, {IsComplexType}";
+        }
+
+        public readonly struct ExtensionMethodInfo
+        {
+            public string FullyQualifiedContainerName { get; }
+
+            public string Name { get; }
+
+            public ExtensionMethodInfo(string fullyQualifiedContainerName, string name)
+            {
+                FullyQualifiedContainerName = fullyQualifiedContainerName;
+                Name = name;
+            }
+        }
+
+        private sealed class ParameterTypeInfoProvider : ISignatureTypeProvider<ParameterTypeInfo, object>
+        {
+            public static readonly ParameterTypeInfoProvider Instance = new ParameterTypeInfoProvider();
+
+            private static ParameterTypeInfo ComplexInfo
+                => new ParameterTypeInfo(string.Empty, isComplex: true);
+
+            public ParameterTypeInfo GetPrimitiveType(PrimitiveTypeCode typeCode)
+                => new ParameterTypeInfo(typeCode.ToString(), isComplex: false);
+
+            public ParameterTypeInfo GetArrayType(ParameterTypeInfo elementType, ArrayShape shape) => ComplexInfo;
+
+            public ParameterTypeInfo GetSZArrayType(ParameterTypeInfo elementType) => ComplexInfo;
+
+            public ParameterTypeInfo GetGenericInstantiation(ParameterTypeInfo genericType, ImmutableArray<ParameterTypeInfo> typeArguments)
+                => genericType.IsComplexType
+                    ? ComplexInfo
+                    : new ParameterTypeInfo(genericType.Name, isComplex: false);
+
+            public ParameterTypeInfo GetByReferenceType(ParameterTypeInfo elementType) => ComplexInfo;
+
+            public ParameterTypeInfo GetFunctionPointerType(MethodSignature<ParameterTypeInfo> signature) => ComplexInfo;
+
+            public ParameterTypeInfo GetGenericMethodParameter(object genericContext, int index) => ComplexInfo;
+
+            public ParameterTypeInfo GetGenericTypeParameter(object genericContext, int index) => ComplexInfo;
+
+            public ParameterTypeInfo GetModifiedType(ParameterTypeInfo modifier, ParameterTypeInfo unmodifiedType, bool isRequired) => ComplexInfo;
+
+            public ParameterTypeInfo GetPinnedType(ParameterTypeInfo elementType) => ComplexInfo;
+
+            public ParameterTypeInfo GetPointerType(ParameterTypeInfo elementType) => ComplexInfo;
```

Comment:
consider grouping all the ones that unilaterally produce a ComplexInfo.

Reply:
comment still applies.

Reply:
done